mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala

```diff
@@ -547,6 +550,45 @@ class ALSSuite
       ALS.train(ratings)
     }
   }
+
+  test("recommendForAllUsers") {
+    val numUsers = 20
+    val numItems = 40
+    val numRecs = 5
+    val (training, test) = genExplicitTestData(numUsers, numItems, rank = 2, noiseStd = 0.01)
+    val topItems =
+      testALS(training, test, maxIter = 4, rank = 2, regParam = 0.01, targetRMSE = 0.03)
```

Comment:
Seems wasteful to compute and check a model here, and it doesn't really test that the predictions are what we expect them to be.

We can construct an `ALSModel` with known factors and check the predictions are as expected (it's just a matrix multiply). See `MatrixFactorizationModelSuite` and tests in #12574 (based on those) for example.